File: programmers/level1/최대공약수와 최소공배수/solution.js
/**
 *
 * @param {number} num1
 * @param {number} num2
 * @returns {number}
 */
const getGCD = (num1, num2) => {
  // 유클리드 호제법 사용
  return num2 > 0 ? getGCD(num2, num1 % num2) : num1;
};

function solution(n, m) {
  const gcd = getGCD(n, m);
  return [gcd, n * (m / gcd)];
}

File: programmers/level1/최대공약수와 최소공배수/solution.kt
class Solution {
    fun solution(a: Int, b: Int): IntArray {
        val gcd = getGCD(a,b)
        return intArrayOf(gcd, a * (b / gcd))
    }

    fun getGCD(a: Int, b: Int) : Int = if (b > 0) getGCD(b, a % b) else a
}